### PR TITLE
Update debug mode logging to include hero role

### PR DESCRIPTION
**Pull request by idshibanov**
Added AI hero role to the debug output and reformatted the rest of it to make it a bit more concise.

Also the "has no parent tile" warning is my personal annoyance since it happens frequently and pollutes the log.

### CODE REVIEW

File: src/fheroes2/heroes/heroes.cpp [hunk 1]
```diff
@@ -107,6 +107,26 @@ namespace
...
+        case Heroes::Role::COURIER:
+            return _( "Courier" );
+        case Heroes::Role::HUNTER:
+            return _( "Hunter" );
+        case Heroes::Role::FIGHTER:
+            return _( "Fighter" );
+        case Heroes::Role::CHAMPION:
+            return _( "Champion" );
+        default:
+            break;
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes.cpp:125]: I suggest to add
```cpp
// Did you add a new AI hero role? Add the appropriate logic for it!
assert( 0 );
```
here to notify a developer that they miss something.

File: src/fheroes2/heroes/heroes.cpp [hunk 2]
```diff
@@ -107,6 +107,26 @@ namespace
 
         return result;
     }
+
+    std::string GetHeroRoleString( const Heroes & hero )
+    {
+        switch ( hero.getAIRole() ) {
+        case Heroes::Role::SCOUT:
+            return _( "Scout" );
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes.cpp:115]: We display these strings not for users but for developers. In this case do we need to have translation for them? If the answer is still yes I would suggest to add a prefix for them. For example, `_( "ai-hero-role|Scout" );`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov , I put several comment here. Could you please check them?

File: src/fheroes2/heroes/heroes.cpp (hunk 2 above)
Comment (idshibanov) [reply src/fheroes2/heroes/heroes.cpp:115]: Good catch. I don't think we need a translation here, I'll get rid of it.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@idshibanov , thanks a lot for these changes!

**Pull request merged**